src/Intl/Transliterator/bootstrap.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Symfony\Polyfill\Intl\Transliterator as p;
+
+if (!function_exists('transliterator_create')) {
+    function transliterator_create($id, $direction = null) { return p\Transliterator::create($id, $direction); }
+    function transliterator_create_from_rules($rules, $direction = null) { return p\Transliterator::createFromRules($rules, $direction); }
+    function transliterator_list_ids() { return p\Transliterator::listIDs(); }
+    function transliterator_create_inverse() { return p\Transliterator::createInverse(); }
+    function transliterator_transliterate(p\Transliterator $trans, $subject, $start = null, $end = null) { return $trans->transliterate($subject, $start, $end); }
+    function transliterator_get_error_code(p\Transliterator $trans) { return $trans->getErrorCode(); }
+    function transliterator_get_error_message(p\Transliterator $trans) { return $trans->getErrorMessage(); }
```

Comment:
this should type-hint for \Transliterator directly, and stubs should exist for the class,
see `src/Intl/Normalizer/Resources/stubs/Normalizer.php` and the classmap in the corresponding composer.json

Reply:
Now I added a stubs file, but this stubs are not used in other "bootstrap.php"-files? Should we use it here? e.g. ```src/Intl/Normalizer/bootstrap.php```

Reply:
the file should not be required in bootstrap.php. It should be referenced in the composer autoload config

Reply:
@stof now I use the autoloader + type-hint for \Transliterator directly (in this bootstrap.php) but it's not used this way in other "bootstrap.php" file e.g. 

src/Intl/Normalizer/bootstrap.php


```
use Symfony\Polyfill\Intl\Normalizer as p;

if (!function_exists('normalizer_is_normalized')) {
    function normalizer_is_normalized($s, $form = p\Normalizer::NFC) { return p\Normalizer::isNormalized($s, $form); }
    function normalizer_normalize($s, $form = p\Normalizer::NFC) { return p\Normalizer::normalize($s, $form); }
}
```

Reply:
that's not a typehint. That's a constant access. So what's important here is the value being used. For the typehint, using the `Symfony\Polyfill\Intl\Normalizer` class rather than the class from the global namespace produces a different function signature.

Reply:
@stof ok, understood and thanks for the explanation. Is this code correct now?

```php
if (!function_exists('transliterator_create')) {
    function transliterator_create($id, $direction = null) { return \Transliterator::create($id, $direction); }
    function transliterator_create_from_rules($rules, $direction = null) { return \Transliterator::createFromRules($rules, $direction); }
    function transliterator_list_ids() { return \Transliterator::listIDs(); }
    function transliterator_create_inverse() {return \Transliterator::createInverse(); }
    function transliterator_transliterate(\Transliterator $trans, $subject, $start = null, $end = null) { return $trans->transliterate($subject, $start, $end); }
    function transliterator_get_error_code(\Transliterator $trans) { return $trans->getErrorCode(); }
    function transliterator_get_error_message(\Transliterator $trans) { return $trans->getErrorMessage(); }
}
```

Reply:
yes, looks good to me.